diff --git a/src/Makefile.am b/src/Makefile.am
index 011a78c..322c7b0 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -96,6 +96,7 @@ noinst_HEADERS = \
 	ol_player_rhythmbox.h \
 	ol_player_songbird.h \
 	ol_player_vlc.h \
+	ol_player_deadbeef.h \
 	ol_player_xmms2.h \
 	ol_player_mpris2.h \
 	ol_scroll_module.h \
@@ -149,6 +150,7 @@ osd_lyrics_SOURCES = \
 	ol_player_deciber.c \
 	ol_player_gmusicbrowser.c \
 	ol_player_vlc.c \
+	ol_player_deadbeef.c \
 	ol_player_mpris2.c \
 	ol_elapse_emulator.c \
 	ol_md5.c \
diff --git a/src/ol_player.c b/src/ol_player.c
index 4389cad..b740320 100644
--- a/src/ol_player.c
+++ b/src/ol_player.c
@@ -50,6 +50,7 @@
 #include "ol_player_juk.h"
 #include "ol_player_muine.h"
 #include "ol_player_mpris2.h"
+#include "ol_player_deadbeef.h"
 
 static GArray *players = NULL;
 
@@ -88,6 +89,7 @@ ol_player_init ()
     ol_player_register (ol_player_juk_get ());
     ol_player_register (ol_player_muine_get ());
 
+    ol_player_register (ol_player_deadbeef_get());
 
   }
 }
diff --git a/src/ol_player_deadbeef.c b/src/ol_player_deadbeef.c
new file mode 100644
index 0000000..3e30a30
--- /dev/null
+++ b/src/ol_player_deadbeef.c
@@ -0,0 +1,166 @@
+#include <stdio.h>
+#include <sys/time.h>
+#include <dbus/dbus-glib.h>
+#include "ol_player_deadbeef.h"
+#include "ol_utils.h"
+#include "ol_utils_dbus.h"
+#include "ol_player_mpris.h"
+#include "ol_debug.h"
+
+static const char *SERVICE = "org.mpris.deadbeef";
+static const char *icon_paths[] = {
+  "/usr/share/icons/hicolor/48x48/apps/deadbeef.png",
+  "/usr/local/share/icons/hicolor/48x48/apps/deadbeef.png",
+};
+
+struct timeval begin_time;
+
+static OlPlayerMpris *mpris = NULL;
+
+static OlPlayerMpris* ol_player_deadbeef_get_mpris ();
+
+static gboolean ol_player_deadbeef_get_music_info (OlMusicInfo *info);
+static gboolean ol_player_deadbeef_get_played_time (int *played_time);
+static gboolean ol_player_deadbeef_get_music_length (int *len);
+static gboolean ol_player_deadbeef_get_activated ();
+static enum OlPlayerStatus ol_player_deadbeef_get_status ();
+static int ol_player_deadbeef_get_capacity ();
+static gboolean ol_player_deadbeef_play ();
+static gboolean ol_player_deadbeef_pause ();
+static gboolean ol_player_deadbeef_stop ();
+static gboolean ol_player_deadbeef_prev ();
+static gboolean ol_player_deadbeef_next ();
+static gboolean ol_player_deadbeef_seek (int pos_ms);
+static const char *_get_icon_path ();
+
+static OlPlayerMpris*
+ol_player_deadbeef_get_mpris ()
+{
+  if (mpris == NULL)
+  {
+    mpris = ol_player_mpris_new (SERVICE);
+  }
+  return mpris;
+}
+
+static gboolean
+ol_player_deadbeef_get_music_info (OlMusicInfo *info)
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_get_music_info (mpris, info);
+}
+
+static gboolean
+ol_player_deadbeef_get_played_time (int *played_time)
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  /* if (!ol_player_mpris_get_played_time (mpris, played_time)) */
+  /*   return FALSE; */
+  /* *played_time = ol_player_deadbeef_get_real_ms (*played_time); */
+  /* return TRUE; */
+  return ol_player_mpris_get_played_time (mpris, played_time);
+}
+
+static gboolean
+ol_player_deadbeef_get_music_length (int *len)
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_get_music_length (mpris, len);
+}
+
+static gboolean
+ol_player_deadbeef_get_activated ()
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_get_activated (mpris);
+}
+
+static enum OlPlayerStatus
+ol_player_deadbeef_get_status ()
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_get_status (mpris);
+}
+
+static int
+ol_player_deadbeef_get_capacity ()
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_get_capacity (mpris);
+}
+
+static gboolean
+ol_player_deadbeef_play ()
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_play (mpris);
+}
+
+static gboolean
+ol_player_deadbeef_pause ()
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_pause (mpris);
+}
+
+static gboolean
+ol_player_deadbeef_stop ()
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_stop (mpris);
+}
+
+static gboolean
+ol_player_deadbeef_prev ()
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_prev (mpris);
+}
+
+static gboolean
+ol_player_deadbeef_next ()
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_next (mpris);
+}
+
+static gboolean
+ol_player_deadbeef_seek (int pos_ms)
+{
+  OlPlayerMpris *mpris = ol_player_deadbeef_get_mpris ();
+  return ol_player_mpris_seek (mpris, pos_ms);
+}
+
+static const char *
+_get_icon_path ()
+{
+  int i;
+  for (i = 0; i < ol_get_array_len (icon_paths); i++)
+  {
+    if (ol_path_is_file (icon_paths[i]))
+      return icon_paths[i];
+  }
+  return NULL;
+}
+
+struct OlPlayer*
+ol_player_deadbeef_get ()
+{
+  ol_log_func ();
+  struct OlPlayer *controller = ol_player_new ("DeaDBeeF");
+  ol_player_set_cmd (controller, "deadbeef");
+  controller->get_music_info = ol_player_deadbeef_get_music_info;
+  controller->get_activated = ol_player_deadbeef_get_activated;
+  controller->get_played_time = ol_player_deadbeef_get_played_time;
+  controller->get_music_length = ol_player_deadbeef_get_music_length;
+  controller->get_status = ol_player_deadbeef_get_status;
+  controller->get_capacity = ol_player_deadbeef_get_capacity;
+  controller->play = ol_player_deadbeef_play;
+  controller->pause = ol_player_deadbeef_pause;
+  controller->stop = ol_player_deadbeef_stop;
+  controller->prev = ol_player_deadbeef_prev;
+  controller->next = ol_player_deadbeef_next;
+  controller->seek = ol_player_deadbeef_seek;
+  controller->get_icon_path = _get_icon_path;
+  return controller;
+}
diff --git a/src/ol_player_deadbeef.h b/src/ol_player_deadbeef.h
new file mode 100644
index 0000000..ee0f00d
--- /dev/null
+++ b/src/ol_player_deadbeef.h
@@ -0,0 +1,8 @@
+#ifndef _OL_PLAYER_DEADBEEF_H_
+#define _OL_PLAYER_DEADBEEF_H_
+
+#include "ol_player.h"
+
+struct OlPlayer* ol_player_deadbeef_get ();
+
+#endif /* _OL_PLAYER_DEADBEEF_H_ */
